rest/route/agent.go

```diff
@@ -353,14 +354,128 @@ func (h *getExpansionsHandler) Run(ctx context.Context) gimlet.Responder {
 		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "getting GitHub OAuth token"))
 	}
 
-	e, err := model.PopulateExpansions(ctx, h.settings, t, foundHost, oauthToken)
+	e, err := model.PopulateExpansions(t, foundHost, oauthToken)
 	if err != nil {
 		return gimlet.NewJSONInternalErrorResponse(err)
 	}
+	v, err := model.VersionFindOneId(t.Version)
+	if err != nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "finding version"))
+	}
+	if v == nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Errorf("version '%s' not found", t.Version))
+	}
+	bvExpansions, err := model.FindExpansionsForVariant(ctx, h.settings, v, t.BuildVariant)
+	if err != nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "loading build variant expansions"))
+	}
+	e.Update(bvExpansions)
 
 	return gimlet.NewJSONResponse(e)
 }
 
+// GET /task/{task_id}/expansions_and_vars
+type getExpansionsAndVarsHandler struct {
+	settings *evergreen.Settings
+	taskID   string
+	hostID   string
+}
+
+func makeGetExpansionsAndVars(settings *evergreen.Settings) gimlet.RouteHandler {
+	return &getExpansionsAndVarsHandler{
+		settings: settings,
+	}
+}
+
+func (h *getExpansionsAndVarsHandler) Factory() gimlet.RouteHandler {
+	return &getExpansionsAndVarsHandler{
+		settings: h.settings,
+	}
+}
+
+func (h *getExpansionsAndVarsHandler) Parse(ctx context.Context, r *http.Request) error {
+	if h.taskID = gimlet.GetVars(r)["task_id"]; h.taskID == "" {
+		return errors.New("missing task ID")
+	}
+	h.hostID = r.Header.Get(evergreen.HostHeader)
+	podID := r.Header.Get(evergreen.PodHeader)
+	if h.hostID == "" && podID == "" {
+		return errors.New("missing both host and pod ID")
```

Comment:
I think this is unnecessary since the `requirePodOrHost` middleware set on this route already handles this case

Reply:
I just copy-pasted this from the existing expansions/vars route. It is unnecessary because of the auth middleware, but I kind of prefer to keep it around just because 1. the auth middleware is not actually important here, it simply happens to conveniently bump the last communication time, and 2. the middleware set for the route is independent of the route handler itself so I'd prefer the handler validate its own inputs. For example, if we changed the middleware behavior or removed the host/pod auth middleware, this route handler would still want to validate the correctness of the required request inputs during parsing.